stacks/spica/src/bucket/services/bucket.service.ts

```diff
@@ -13,11 +13,16 @@ import * as fromBucket from "../state/bucket.reducer";
 
 @Injectable()
 export class BucketService {
+  patchHttpOptions: object;
   constructor(
     private http: HttpClient,
     private store: Store<fromBucket.State>,
     private preference: PreferencesService
-  ) {}
+  ) {
+    this.patchHttpOptions = {
+      headers: new HttpHeaders().set("Content-Type", "application/merge-patch+json")
+    };
```

Comment:
Why we don't move these lines into the patchIndexes method?

Reply:
Drag-drop events call this method for all buckets to patch their indexes, so patchIndexes method shouldn't create a new HttpHeaders object for all calls, isn't it?

Reply:
As a matter of fact, it should, because we want to not have "states in classes" as much as possible.

Reply:
Since the client application is distributed out of the box, we don't have to worry about a simple allocation.